File: 灵茶/滑动窗口与双指针/1343.java
class Solution {

    public int numOfSubarrays(int[] arr, int k, int threshold) {
        int result = 0;
        int sum = 0;
        for (int i = 0; i < arr.length; i++) {
            sum += arr[i];

            if (i >= k - 1) {
                if (sum >= k * threshold) {
                    result++;
                }
                sum -= arr[i - (k - 1)];

            }
        }
        return result;
    }
}

File: 灵茶/滑动窗口与双指针/定长窗口/1343.py
class Solution:
    def numOfSubarrays(self, arr: List[int], k: int, threshold: int) -> int:
        sum = 0
        ans = 0
        for index, num in enumerate(arr):
            sum += num

            if index < k - 1:
                continue
            
            if sum >= threshold * k:
                ans += 1
            
            sum -= arr[index - k + 1]
        return ans
